Improve documentation with templates (#366)
SUSE/ha-sap-terraform-deployments: doc/deployment-templates.md

> # Deployement templates:

Reviewer: typo

```suggestion
# Deployment templates
```

Reviewer: and this one too

> This doc is an invetory of common functional templates, use-cases of the deployment project.

Reviewer: typo and wording

```suggestion
This document is an inventory of functional templates for common use-cases of the deployment project.
```

Reviewer: I think you missed this one!

> This examples were used with KVM libvrt but the other providers will have the same vars or differ not significantly. 

Reviewer: typo and wording

```suggestion
The following examples refer to the libvirt Terraform provider, but they may be used for the other providers as well: there are no significant differences for the most common settings.
```

- Hana cluster (the simplest one)
- Monitoring hana cluster
- ISCSI Server
- NETWEAVER

Not implementd yet:
- NETWEAVER Monitoring (need implementation)

Additionally you need to set up pillars. In dev-mode we use mostly `automatic`. 
See https://github.com/SUSE/ha-sap-terraform-deployments/tree/master/pillar_examples#pillar-examples for more details.

# hana cluster

```
qemu_uri = "qemu+ssh://MYUSER@MYSTEM/system"
base_image = "URL_TO_IMAGE"

> iprange = "192.168.210.0/24"

Reviewer 2: the iprange value (210) does not fit to your chosen host_ips (110) in line 26

in all of your examples

Author: thx

Author: adapted doc accordingly

hana_inst_media = "PATH TO INST_MEDIA"
host_ips = ["192.168.110.19", "192.168.110.20"]
reg_code = "MY_REG_CODE"
reg_email = "MY_EMAIL"
ha_sap_deployment_repo = "http://download.opensuse.org/repositories/network:/ha-clustering:/Factory/SLE_15/"
shared_storage_type = "shared-disk"
storage_pool           = "terraform"
```

# monitoring hana cluster

> To monitoring ha_cluster you need only 2 vars more then hana_cluster deployment.

Reviewer: I don't think there is any need to use snake_case here.

```suggestion
To monitoring the HANA cluster need only 2 vars more than the simple HANA deployment.
```
